driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -724,6 +727,23 @@ public Builder withPort(int port) {
       return this;
     }
 
+    /**
+     * Enables host port discovery using the system.peers_v2 table added in Cassandra 4.0 (via
+     * CASSANDRA-7544). This enables running multiple Cassandra
+     *
+     * <p>Use of this option only works for clusters running version Cassandra 4.0 or newer. Using
+     * it with an older version of Cassandra will likely cause initialization to fail.
+     *
+     * <p>When using this option configuration provided via {@link #withPort(int)} is unused as
+     * ports are resolved from Cassandra.
+     *
+     * @return this builder.
+     */
+    public Builder allowHostPortDiscovery() {
```

Comment:
Why expose this? Can't we simply detect the presence of the new peers_v2 table? Or is it because if the cluster has some nodes with peers_v2 and others without, that might be a too complex situation to handle?

Reply:
I explain my rationale for this here: https://github.com/datastax/java-driver/pull/1065#discussion_r206263173.  I originally tried peers_v2 first, then peers if the table doesn't exist (like 4.x does).   But thinking about it more I wasn't sure if adding another query in a minor version was the right thing to do, so I thought maybe making this an Opt-in feature on 3.x would be the right way to go.

Reply:
So are we doing the fallback to `system.peers` at all anymore? The PR description says so...

Reply:
I was erring on the side of caution by making this an 'opt-in' feature.  If you don't see that as too risky, I will revert it back to my initial implementation which mirrors the approach we took with 4.x.    With regards to not falling back to `system.peers`, my thought process was that if you explicitly opted in using `allowHostPortDiscovery()`, it didn't seem the backing off to peers was the best idea, since you expressing that you want to utilize the capability which is not being provided by the server.  If i revert to not making this opt-in that point is moot though.